src/Microsoft.Dnx.Tooling/Publish/PublishProject.cs

```diff
@@ -352,95 +352,94 @@ public bool PostProcess(PublishRoot root)
-            var format = new LockFileFormat();
-            root.LockFile = format.Read(lockFilePath);
+            root.PublishedLockFile = new LockFileFormat().Read(lockFilePath);
 
-            var keep = new HashSet<string>();
+            var filesToKeep = new HashSet<string>();
+            var filesToRemove = new HashSet<string>();
 
-            foreach (var target in root.LockFile.Targets)
+            foreach (var target in root.PublishedLockFile.Targets)
             {
                 foreach (var library in target.Libraries)
                 {
-                    var packagesDir = resolver.GetInstallPath(library.Name, library.Version);
-                    var manifest = resolver.GetManifestFilePath(library.Name, library.Version);
-
-                    keep.Add(manifest);
+                    var packageDir = resolver.GetInstallPath(library.Name, library.Version);
+                    
+                    if (library.Name != root.MainProjectName && string.Equals(library.Type, LibraryTypes.Package, StringComparison.OrdinalIgnoreCase))
+                    {
+                        filesToRemove.Add(resolver.GetHashPath(library.Name, library.Version));
+                        filesToRemove.Add(resolver.GetPackageFilePath(library.Name, library.Version));
+                        filesToRemove.AddRange(Directory.EnumerateFiles(packageDir, $"{library.Name}.xml", SearchOption.AllDirectories));
+                    }
 
                     foreach (var path in library.RuntimeAssemblies)
                     {
-                        keep.Add(CombinePath(packagesDir, path));
+                        filesToKeep.Add(Path.Combine(packageDir, path));
                     }
-
                     foreach (var path in library.CompileTimeAssemblies)
                     {
-                        keep.Add(CombinePath(packagesDir, path));
+                        filesToKeep.Add(Path.Combine(packageDir, path));
                     }
-
                     foreach (var path in library.NativeLibraries)
                     {
-                        keep.Add(CombinePath(packagesDir, path));
+                        filesToKeep.Add(Path.Combine(packageDir, path));
                     }
-
                     foreach (var path in library.ResourceAssemblies)
                     {
-                        keep.Add(CombinePath(packagesDir, path));
+                        filesToKeep.Add(Path.Combine(packageDir, path));
                     }
+                }
+            }
 
-                    foreach (var specialFolder in specialFolders)
-                    {
-                        var specialFolderPath = CombinePath(packagesDir, specialFolder);
+            foreach (var target in root.MainProjectLockFile.Targets.Where(projectTarget =>
+                !root.PublishedLockFile.Targets.Any(publishTarget =>
+                projectTarget.TargetFramework == publishTarget.TargetFramework)))
```

Comment:
Difficult to do lockfile based `RID` filtering for now since the published project is not restored with the runtimes specified by publish. Could potentially revisit this once publish is refactored to be based on lock file.